src/main/java/com/melon/portfoliomanager/services/AssetService.java

```diff
@@ -0,0 +1,60 @@
+package com.melon.portfoliomanager.services;
+
+
+import com.melon.portfoliomanager.dtos.TransactionDto;
+import com.melon.portfoliomanager.exceptions.NoSuchUserException;
+import com.melon.portfoliomanager.models.PortfolioItem;
+import com.melon.portfoliomanager.models.Transaction;
+import com.melon.portfoliomanager.models.TransactionType;
+import com.melon.portfoliomanager.models.User;
+import com.melon.portfoliomanager.repositories.PortfolioItemRepository;
+import com.melon.portfoliomanager.repositories.TransactionRepository;
+import com.melon.portfoliomanager.repositories.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class AssetService {
+
+    private TransactionRepository transactionRepository;
+    private PortfolioItemRepository portfolioItemRepository;
+    private UserRepository userRepository;
+
+
+    @Autowired
+    public AssetService(TransactionRepository transactionRepository, PortfolioItemRepository portfolioItemRepository, UserRepository userRepository) {
+        this.transactionRepository = transactionRepository;
+        this.portfolioItemRepository = portfolioItemRepository;
+        this.userRepository = userRepository;
+    }
+
+    public void buyStock(TransactionDto transactionDto) {
+
+        List<User> userList = userRepository.findByUsername(transactionDto.getUsername());
+
+        if (userList.isEmpty()) {
+            throw new NoSuchUserException("There is no user with such username. Enter a valid username.");
+        }
+
+        User user = userList.get(0);
```

Comment:
Code for writing in Transactions database table can also be extracted in some private method.

Reply:
fixed that.